compat/v1/compat.go

```diff
@@ -0,0 +1,323 @@
+			Output: Output{
+				Outpoint: Outpoint{
+					Hash:  bchOutput.Hash.String(),
+					Index: bchOutput.Index.String(),
+				},
+				Pubkeyscript: bchOutput.PubKeyScript.String(),
+				Value:        bchOutput.Value.String(),
+			},
+			Pubkey: bchShard.PubKey.String(),
+		}
+	}
+
+	digibyteS, ok := state[string(multichain.DigiByte.NativeAsset())]
+	if ok {
+
+		if len(digibyteS.Shards) == 0 {
+			return StateResponse{},
+				fmt.Errorf("No Digibyte Shards")
+		}
+
+		dgbShard := digibyteS.Shards[0]
+
+		var dgbOutput engine.XStateShardUTXO
+		if err := pack.Decode(&dgbOutput, dgbShard.State); err != nil {
+			return StateResponse{},
+				fmt.Errorf("Failed to unmarshal digibyte shard state: %v", err)
+		}
+
+		stateResponse.Digibyte = UTXOState{
+			Address:           dgbShard.Shard.String(),
+			Dust:              digibyteS.DustAmount.String(),
+			Gascap:            digibyteS.GasCap.String(),
+			Gaslimit:          digibyteS.GasLimit.String(),
+			Gasprice:          digibyteS.GasPrice.String(),
+			Latestchainheight: digibyteS.LatestHeight.String(),
+			Minimumamount:     digibyteS.MinimumAmount.String(),
+			Output: Output{
+				Outpoint: Outpoint{
+					Hash:  dgbOutput.Hash.String(),
+					Index: dgbOutput.Index.String(),
+				},
+				Pubkeyscript: dgbOutput.PubKeyScript.String(),
+				Value:        dgbOutput.Value.String(),
+			},
+			Pubkey: dgbShard.PubKey.String(),
+		}
+
+	}
+
+	dogecoinS, ok := state[string(multichain.DigiByte.NativeAsset())]
+	if ok {
+
+		if len(dogecoinS.Shards) == 0 {
+			return StateResponse{},
+				fmt.Errorf("No Dogecoin Shards")
+		}
+
+		dogeShard := dogecoinS.Shards[0]
+
+		var dogeOutput engine.XStateShardUTXO
+		if err := pack.Decode(&dogeOutput, dogeShard.State); err != nil {
+			return StateResponse{},
+				fmt.Errorf("Failed to unmarshal dogecoin shard state: %v", err)
+		}
+
+		stateResponse.Dogecoin = UTXOState{
+			Address:           dogeShard.Shard.String(),
+			Dust:              dogecoinS.DustAmount.String(),
+			Gascap:            dogecoinS.GasCap.String(),
+			Gaslimit:          dogecoinS.GasLimit.String(),
+			Gasprice:          dogecoinS.GasPrice.String(),
+			Latestchainheight: dogecoinS.LatestHeight.String(),
+			Minimumamount:     dogecoinS.MinimumAmount.String(),
+			Output: Output{
+				Outpoint: Outpoint{
+					Hash:  dogeOutput.Hash.String(),
+					Index: dogeOutput.Index.String(),
+				},
+				Pubkeyscript: dogeOutput.PubKeyScript.String(),
+				Value:        dogeOutput.Value.String(),
+			},
+			Pubkey: dogeShard.PubKey.String(),
+		}
+
+	}
+
+	terraS, ok := state[string(multichain.Terra.NativeAsset())]
+	if ok {
+
+		if len(terraS.Shards) == 0 {
+			return StateResponse{},
+				fmt.Errorf("No Terra Shards")
+		}
+
+		lunaShard := terraS.Shards[0]
+
+		var lunaOutput engine.XStateShardAccount
+		if err := pack.Decode(&lunaOutput, lunaShard.State); err != nil {
+			return StateResponse{},
+				fmt.Errorf("Failed to unmarshal terra shard state: %v", err)
+		}
```

Comment:
```suggestion
		if len(bitcoinS.Shards) == 0 {
			return StateResponse{},
				fmt.Errorf("no bitcoin shards")
		}

		btcShard := bitcoinS.Shards[0]

		var btcOutput engine.XStateShardUTXO
		if err := pack.Decode(&btcOutput, btcShard.State); err != nil {
			return StateResponse{},
				fmt.Errorf("failed to unmarshal bitcoin shard state: %v", err)
		}

		stateResponse.Bitcoin = UTXOState{
			Address:           btcShard.Shard.String(),
			Dust:              bitcoinS.DustAmount.String(),
			Gascap:            bitcoinS.GasCap.String(),
			Gaslimit:          bitcoinS.GasLimit.String(),
			Gasprice:          bitcoinS.GasPrice.String(),
			Latestchainheight: bitcoinS.LatestHeight.String(),
			Minimumamount:     bitcoinS.MinimumAmount.String(),
			Output: Output{
				Outpoint: Outpoint{
					Hash:  btcOutput.Hash.String(),
					Index: btcOutput.Index.String(),
				},
				Pubkeyscript: btcOutput.PubKeyScript.String(),
				Value:        btcOutput.Value.String(),
			},
			Pubkey: btcShard.PubKey.String(),
		}

	}

	zcashS, ok := state[string(multichain.Zcash.NativeAsset())]
	if ok {
		if len(zcashS.Shards) == 0 {
			return StateResponse{},
				fmt.Errorf("no zcash shards")
		}

		zecShard := zcashS.Shards[0]

		var zecOutput engine.XStateShardUTXO
		if err := pack.Decode(&zecOutput, zecShard.State); err != nil {
			return StateResponse{},
				fmt.Errorf("failed to unmarshal zcash shard state: %v", err)
		}

		stateResponse.Zcash = UTXOState{
			Address:           zecShard.Shard.String(),
			Dust:              zcashS.DustAmount.String(),
			Gascap:            zcashS.GasCap.String(),
			Gaslimit:          zcashS.GasLimit.String(),
			Gasprice:          zcashS.GasPrice.String(),
			Latestchainheight: zcashS.LatestHeight.String(),
			Minimumamount:     zcashS.MinimumAmount.String(),
			Output: Output{
				Outpoint: Outpoint{
					Hash:  zecOutput.Hash.String(),
					Index: zecOutput.Index.String(),
				},
				Pubkeyscript: zecOutput.PubKeyScript.String(),
				Value:        zecOutput.Value.String(),
			},
			Pubkey: zecShard.PubKey.String(),
		}
	}

	bitcoinCashS, ok := state[string(multichain.BitcoinCash.NativeAsset())]
	if ok {

		if len(bitcoinCashS.Shards) == 0 {
			return StateResponse{},
				fmt.Errorf("no bitcoin cash shards")
		}

		bchShard := bitcoinCashS.Shards[0]

		var bchOutput engine.XStateShardUTXO
		if err := pack.Decode(&bchOutput, bchShard.State); err != nil {
			return StateResponse{},
				fmt.Errorf("failed to unmarshal bitcoinCash shard state: %v", err)
		}

		stateResponse.Bitcoincash = UTXOState{
			Address:           bchShard.Shard.String(),
			Dust:              bitcoinCashS.DustAmount.String(),
			Gascap:            bitcoinCashS.GasCap.String(),
			Gaslimit:          bitcoinCashS.GasLimit.String(),
			Gasprice:          bitcoinCashS.GasPrice.String(),
			Latestchainheight: bitcoinCashS.LatestHeight.String(),
			Minimumamount:     bitcoinCashS.MinimumAmount.String(),
			Output: Output{
				Outpoint: Outpoint{
					Hash:  bchOutput.Hash.String(),
					Index: bchOutput.Index.String(),
				},
				Pubkeyscript: bchOutput.PubKeyScript.String(),
				Value:        bchOutput.Value.String(),
			},
			Pubkey: bchShard.PubKey.String(),
		}
	}

	digibyteS, ok := state[string(multichain.DigiByte.NativeAsset())]
	if ok {

		if len(digibyteS.Shards) == 0 {
			return StateResponse{},
				fmt.Errorf("no digibyte shards")
		}

		dgbShard := digibyteS.Shards[0]

		var dgbOutput engine.XStateShardUTXO
		if err := pack.Decode(&dgbOutput, dgbShard.State); err != nil {
			return StateResponse{},
				fmt.Errorf("failed to unmarshal digibyte shard state: %v", err)
		}

		stateResponse.Digibyte = UTXOState{
			Address:           dgbShard.Shard.String(),
			Dust:              digibyteS.DustAmount.String(),
			Gascap:            digibyteS.GasCap.String(),
			Gaslimit:          digibyteS.GasLimit.String(),
			Gasprice:          digibyteS.GasPrice.String(),
			Latestchainheight: digibyteS.LatestHeight.String(),
			Minimumamount:     digibyteS.MinimumAmount.String(),
			Output: Output{
				Outpoint: Outpoint{
					Hash:  dgbOutput.Hash.String(),
					Index: dgbOutput.Index.String(),
				},
				Pubkeyscript: dgbOutput.PubKeyScript.String(),
				Value:        dgbOutput.Value.String(),
			},
			Pubkey: dgbShard.PubKey.String(),
		}

	}

	dogecoinS, ok := state[string(multichain.DigiByte.NativeAsset())]
	if ok {

		if len(dogecoinS.Shards) == 0 {
			return StateResponse{},
				fmt.Errorf("no dogecoin shards")
		}

		dogeShard := dogecoinS.Shards[0]

		var dogeOutput engine.XStateShardUTXO
		if err := pack.Decode(&dogeOutput, dogeShard.State); err != nil {
			return StateResponse{},
				fmt.Errorf("failed to unmarshal dogecoin shard state: %v", err)
		}

		stateResponse.Dogecoin = UTXOState{
			Address:           dogeShard.Shard.String(),
			Dust:              dogecoinS.DustAmount.String(),
			Gascap:            dogecoinS.GasCap.String(),
			Gaslimit:          dogecoinS.GasLimit.String(),
			Gasprice:          dogecoinS.GasPrice.String(),
			Latestchainheight: dogecoinS.LatestHeight.String(),
			Minimumamount:     dogecoinS.MinimumAmount.String(),
			Output: Output{
				Outpoint: Outpoint{
					Hash:  dogeOutput.Hash.String(),
					Index: dogeOutput.Index.String(),
				},
				Pubkeyscript: dogeOutput.PubKeyScript.String(),
				Value:        dogeOutput.Value.String(),
			},
			Pubkey: dogeShard.PubKey.String(),
		}

	}

	terraS, ok := state[string(multichain.Terra.NativeAsset())]
	if ok {

		if len(terraS.Shards) == 0 {
			return StateResponse{},
				fmt.Errorf("no terra shards")
		}

		lunaShard := terraS.Shards[0]

		var lunaOutput engine.XStateShardAccount
		if err := pack.Decode(&lunaOutput, lunaShard.State); err != nil {
			return StateResponse{},
				fmt.Errorf("failed to unmarshal terra shard state: %v", err)
		}
```

Just more error tweaks. On another note, I think we could probably reduce a lot of this code since the logic is the same for each chain, but I'm also not too worried about it because we shouldn't expect this code to change (if anything, it will be removed entirely once we no longer require support for earlier versions).